udata/core/dataset/factories.py

```diff
@@ -58,3 +59,8 @@ class Meta:
 
     id = factory.Sequence(lambda n: '{0}-{1}'.format(faker.word(), n))
     title = factory.LazyAttribute(lambda o: faker.sentence())
+
+
+class UserCreatedDatasetFactory(MongoEngineFactory):
```

Comment:
Given the fact that it needs real objects that you give as parameters and that the factory does not provide any attribute, it might be better to not use a factory.